systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java

```diff
@@ -167,6 +168,28 @@ public static boolean waitForKafkaTopicNotReady(String topicName) {
         return waitForKafkaTopicStatus(kubeClient().getNamespace(), topicName, NotReady);
     }
 
+    public static void waitForTopicConfigContains(String namespaceName, String topicName, Map<String, Object> config) {
+        LOGGER.info("Wait until topic {} contains correct config", topicName);
+        TestUtils.waitFor("Wait for correct config",
+                Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_STATUS_TIMEOUT,
+                () -> KafkaTopicUtils.configsAreEqual(KafkaTopicResource.kafkaTopicClient()
+                        .inNamespace(namespaceName).withName(topicName).get().getSpec().getConfig(), config)
+        );
+        LOGGER.info("Topic {} contains correct config", topicName);
+    }
+
+    public static boolean configsAreEqual(Map<String, Object> actualConf, Map<String, Object> expectedConf) {
+        try {
+            if (expectedConf.size() != actualConf.size()) return false;
+
+            return expectedConf.entrySet().stream()
+                    .allMatch(expected -> expected.getValue().toString().equals(actualConf.get(expected.getKey()).toString()));
+        } catch (NullPointerException e) {
+            LOGGER.error(e.getMessage());
+            return false;
+        }
```

Comment:
Maybe you can filter the actualConf with expectedConf and then just check if the sizes are same? I guess you will be able to get rid of the try catch and no NPE will be thrown.

Reply:
Dou you mean filter by conditions == null ? Or is there any other way?